src/normalize/normalizer.rs

```diff
@@ -544,4 +544,56 @@ mod tests {
         };
         assert_eq!(meta, &UserAddrMeta::Binary(expected_binary));
     }
+
+    /// Check that we correctly handle normalization of an address not
+    /// in any executable segment.
+    #[test]
+    fn user_address_normalization_static_maps() {
+        fn read_no_build_id(_path: &Path) -> Result<Option<Vec<u8>>> {
+            Ok(None)
+        }
+
+        let maps = r#"
+55d3195b7000-55d3195b9000 r--p 00000000 00:12 2015701                    /bin/cat
+55d3195b9000-55d3195be000 r-xp 00002000 00:12 2015701                    /bin/cat
+55d3195be000-55d3195c1000 r--p 00007000 00:12 2015701                    /bin/cat
+55d3195c1000-55d3195c2000 r--p 00009000 00:12 2015701                    /bin/cat
+55d3195c2000-55d3195c3000 rw-p 0000a000 00:12 2015701                    /bin/cat
+55d31b4dc000-55d31b4fd000 rw-p 00000000 00:00 0                          [heap]
+7fd5b9c3d000-7fd5b9c5f000 rw-p 00000000 00:00 0
+7fd5b9c5f000-7fd5ba034000 r--p 00000000 00:12 7689533                    /usr/lib/locale/locale-archive
+7fd5ba034000-7fd5ba037000 rw-p 00000000 00:00 0
+7fd5ba037000-7fd5ba059000 r--p 00000000 00:12 2088876                    /lib64/libc.so.6
+7fd5ba059000-7fd5ba1a8000 r-xp 00022000 00:12 2088876                    /lib64/libc.so.6
+7fd5ba1a8000-7fd5ba1fa000 r--p 00171000 00:12 2088876                    /lib64/libc.so.6
+7fd5ba1fa000-7fd5ba1fe000 r--p 001c3000 00:12 2088876                    /lib64/libc.so.6
+7fd5ba1fe000-7fd5ba200000 rw-p 001c7000 00:12 2088876                    /lib64/libc.so.6
+7fd5ba200000-7fd5ba208000 rw-p 00000000 00:00 0
+7fd5ba214000-7fd5ba216000 rw-p 00000000 00:00 0
+7fd5ba216000-7fd5ba217000 r--p 00000000 00:12 2088889                    /lib64/ld-linux-x86-64.so.2
+7fd5ba217000-7fd5ba23c000 r-xp 00001000 00:12 2088889                    /lib64/ld-linux-x86-64.so.2
+7fd5ba23c000-7fd5ba246000 r--p 00026000 00:12 2088889                    /lib64/ld-linux-x86-64.so.2
+7fd5ba246000-7fd5ba248000 r--p 00030000 00:12 2088889                    /lib64/ld-linux-x86-64.so.2
+7fd5ba248000-7fd5ba24a000 rw-p 00032000 00:12 2088889                    /lib64/ld-linux-x86-64.so.2
+7ffe102a2000-7ffe102c4000 rw-p 00000000 00:00 0                          [stack]
+7ffe103f6000-7ffe103fa000 r--p 00000000 00:00 0                          [vvar]
+7ffe103fa000-7ffe103fc000 r-xp 00000000 00:00 0                          [vdso]
+"#;
+
+        let pid = Pid::Slf;
+        let entries = maps::parse_file(maps.as_bytes(), pid);
+        let addrs = [0x7fd5ba1fe000 as Addr];
```

Comment:
why not also check address before any entry and after all entries, just to check edge cases in the logic?

Reply:
Sure, done.